crates/pop-contracts/src/call/metadata.rs

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::errors::Error;
+use contract_extrinsics::ContractArtifacts;
+use contract_transcode::ink_metadata::MessageParamSpec;
+use scale_info::form::PortableForm;
+use std::path::Path;
+
+#[derive(Clone, PartialEq, Eq)]
+/// Describes a contract message.
+pub struct Param {
+	/// The label of the parameter.
+	pub label: String,
+	/// The type name of the parameter.
+	pub type_name: String,
+}
+#[derive(Clone, PartialEq, Eq)]
+/// Describes a contract message.
+pub struct Message {
+	/// The label of the message.
+	pub label: String,
+	/// If the message is allowed to mutate the contract state.
+	pub mutates: bool,
+	/// If the message accepts any `value` from the caller.
+	pub payable: bool,
+	/// The parameters of the deployment handler.
+	pub args: Vec<Param>,
+	/// The message documentation.
+	pub docs: String,
+	/// If the message is the default for off-chain consumers (e.g UIs).
+	pub default: bool,
+}
+
+/// Extracts a list of smart contract messages parsing the metadata file.
+///
+/// # Arguments
+/// * `path` -  Location path of the project.
+pub fn get_messages(path: &Path) -> Result<Vec<Message>, Error> {
```

Comment:
we are expecting path to a manifest, but it would work just fine with a path to the contract bundle. `my_contract.contract`

That way I might not even have to know about the source of the contract, but still interact with it.

Reply:
Solved in https://github.com/r0gue-io/pop-cli/pull/347